### PR TITLE
feat(slice-machine-ui): update PageLayout

### PR DESCRIPTION
**Pull request by bapmrl**
Completes DT-1108 and DT-1527.

This pull request includes the following changes:

1. In `@components/App`:
   1. Stop scanning the URL to decide whether to display the `Navigation` menu. It's now always rendered in the updated `@components/AppLayout` component.
   2. Remove the global `BaseStyles`. It's now applied only where needed by the underlying pages.
   3. Remove the unnecessary `SliceHandler` helper. 
   4. Remove the dynamic import of `@components/AppLayout/Navigation` as it doesn't seem necessary anymore.
2. In `@components/AppLayout`:
   1. Use the new `@src/components/PageLayout` component and always render the `@components/Navigation` menu.
   2. Stop scanning the URL. The `@components/Simulator` page is the only one not using the `@components/AppLayout` component.
3. In `@pages/_app.tsx`:
   1. Import `.css` files first.
   2. Import the new `@styles/globals.css` file to set the `body`'s `height`, `margin`, `overflow`...
4. Update the `@src/components/PageLayout` component to match the Figma design. 
5. Remove the `@components/Container`, `@components/Header`, `@lib/builders/CustomTypeBuilder/Layout/FlexWrapper`, `@lib/builders/SliceBuilder/FlexEditor` and `@src/components/MainContainer` components. They were all kind of defining the app layout.
6. Update the `@src/components/SideNav` component so that its size can grow/shrink depending on the parent.
7. Wrap all `ReactTooltip` elements in the new `@components/ReactTooltipPortal` component to ensure they are drawn above the app.
8. Update the `@prismicio/editor-fields` and `@prismicio/editor-ui` dependencies:
   1. Use the new tokens in the `@src/components/PageLayout` component.
   2. Use the `DropdownMenuItem`'s new `color` prop.
   3. Set `env` to `"prod"` in the `@components/Simulator`'s `editorConfig` object.
   4. Use the new `TooltipProvider` component.

### CODE REVIEW

File: packages/slice-machine/components/App/index.tsx [hunk 1]
```diff
@@ -1,37 +1,22 @@
...
 import useServerState from "@src/hooks/useServerState";
 import { SliceMachineStoreType } from "@src/redux/type";
 import { getLibraries } from "@src/modules/slices";
 import useSMTracker from "@src/hooks/useSMTracker";
 import { useChangelog } from "@src/hooks/useChangelog";
 
-import AppLayout from "../AppLayout";
-
-// TODO: Remove this dynamic import when we have a proper solution to have sprinkles load first
-// Maybe related to https://github.com/vanilla-extract-css/vanilla-extract/pull/1105
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/App/index.tsx:20]: https://github.com/vanilla-extract-css/vanilla-extract/pull/1105
was merged last week :)

Comment (bapmrl) [reply packages/slice-machine/components/App/index.tsx:20]: I've tried it and it didn't solve the CSS issue I had 🙂

File: packages/slice-machine/test/pages/slices.test.tsx [hunk 2]
```diff
@@ -351,9 +352,14 @@ describe("slices", () => {
       fireEvent.change(nameInput, { target: { value: "FooBar" } });
     });
 
-    const submitButton = screen.getByText("Create");
+    const createSliceModal = document.querySelector<HTMLElement>(
```
Comment (MarcMcIntosh) [inline packages/slice-machine/test/pages/slices.test.tsx]: screen has a querySelector to, or is there another reason to use document.querySelector?

File: packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx [hunk 3]
```diff
@@ -46,89 +51,74 @@ export const CustomTypesTablePage: FC<CustomTypesTablePageProps> = ({
...
-              >
+      <ErrorBoundary
+        title="Request failed"
+        description={`An error occurred while fetching your ${customTypesMessages.name(
+          { start: false, plural: true }
+        )}.`}
+        renderError={(error) => (
+          <AppLayout>
+            <AppLayoutContent>
+              <Box alignItems="center" justifyContent="center">
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx:62]: I wonder if there's a design for this?

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx [hunk 4]
```diff
@@ -319,11 +319,16 @@ async function renderCustomTypesTablePage({
...
+      environment: {
+        manifest: { apiEndpoint: "https://foo.cdn.prismic.io/api/v2" },
+      },
+      slices: { libraries: [], remoteSlices: [] },
     },
   };
 
   const renderResults = render(
     <CustomTypesTablePage format={format} />,
+    // @ts-expect-error TS2345: Argument of type '{ preloadedState: { availableCustomTypes: {}; environment: { manifest: { apiEndpoint: string; }; }; slices: { libraries: never[]; remoteSlices: never[]; }; }; }' is not assignable to parameter of type 'Partial<{ preloadedState: Partial<SliceMachineStoreType>; store: Store<SliceMachineStoreType, AnyAction>; } & RenderOptions<...>>'.
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:331]: Seeing a lot of these, has some thing changed?

File: packages/slice-machine/components/AppLayout/index.tsx [hunk 5]
```diff
@@ -1,45 +1,68 @@
...
+  PageLayoutHeader,
+  PageLayoutPane,
+} from "@src/components/PageLayout";
 
-type Props = Readonly<{
-  children?: ReactNode;
-}>;
+export const AppLayout: FC<PropsWithChildren> = ({
+  children,
+  ...otherProps
```
Comment (xrutayisire) [inline packages/slice-machine/components/AppLayout/index.tsx:16]: ❓I never know how to name the this.

Some of us do:
- `...props`
- `...otherProps`
- `...props`
- `...rest`
- `...restProps`
- and even more things

Should we have that in the code guideline, not as a strict thing but more to not think about it?

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx [hunk 6]
```diff
@@ -4,140 +4,61 @@ import VariationModal from "./VariationModal";
...
-          onClose={() => setShowVariationModal(false)}
-          onSubmit={(id, name, copiedVariation) => {
-            copyVariationSlice(id, name, copiedVariation);
+    <>
+      <Flex>
+        <VariationPopover
+          defaultValue={variation}
+          variations={component.model.variations}
+          onNewVariation={() => setShowVariationModal(true)}
+          onChange={(v) =>
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:30]: is `v` short hand for variation?

File: packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx [hunk 7]
```diff
@@ -175,24 +174,24 @@ const SimulatorButton: React.FC<{
...
-            backgroundColor: "#F3F5F7",
-            border: "1px solid #3E3E4826",
-            height: 40,
-            ...(disabled ? { opacity: 0.5, cursor: "not-allowed" } : {}),
-          }}
-        />
+          startIcon={
+            <PlayCircleIcon
+              style={{
+                // TODO(DT-1538): our icons should have a `viewBox` of 24px.
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx:186]: I think most of the icons take SVG props ?

File: packages/slice-machine/components/Simulator/index.tsx [hunk 8]
```diff
@@ -68,6 +68,7 @@ const Simulator: ComponentWithSliceProps = ({ slice, variation }) => {
       embeds: {
         url: endpoints.PrismicOembed,
       },
+      env: "prod",
```
Comment (xrutayisire) [inline packages/slice-machine/components/Simulator/index.tsx:71]: ❓Can you explain me this change?

File: packages/slice-machine/components/Tooltip/TextWithTooltip.tsx [hunk 9]
```diff
@@ -50,17 +52,19 @@ export const TextWithTooltip: React.FC<TextWithTooltipProps> = ({
       </Text>
       {/* eslint-disable-next-line @typescript-eslint/strict-boolean-expressions */}
       {showTooltip && (
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/Tooltip/TextWithTooltip.tsx:54]: will `showToolTip === true` let us remove the `eslint-disable-next-line`?

File: packages/slice-machine/components/Changelog/index.tsx [hunk 10]
```diff
@@ -30,42 +31,36 @@ export default function Changelog() {
...
-        <VersionDetails
-          changelog={changelog}
-          selectedVersion={selectedVersion}
-          packageManager={packageManager}
-        />
-      )}
-    </Flex>
-  ) : (
-    <LoadingPage />
+              {changelog.sliceMachine.versions.length === 0 ||
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/Changelog/index.tsx]: For readability can we have brackets around the conditions?  
`(changelog.sliceMachine.versions.length === 0 || !selectedVersion) ? ...`

**Review by MarcMcIntosh — COMMENTED** [review]
Nice :)
I left a few comments, nothing major though :)

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts [hunk 11]
```diff
@@ -34,15 +34,10 @@ export const root = style([
     backgroundColor: colors.grey2,
     display: "flex",
     flexDirection: "column",
-    height: "100%",
-    minHeight: "100vh",
-    paddingBottom: 32,
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/SideNav/SideNav.css.ts:39]: ⚠ Why removing this padding-bottom here to after add a margin-bottom to last list list `128`?

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts [hunk 12]
```diff
@@ -106,7 +101,7 @@ export const repositoryLinkIcon = style([
     borderWidth: 1,
     cursor: "pointer",
     height: 32,
-    marginRight: 8,
```
Comment (xrutayisire) [inline packages/slice-machine/src/components/SideNav/SideNav.css.ts]: ⚠ Why this change? Is it according to the design?

File: packages/slice-machine/test/pages/custom-types.test.tsx [hunk 13]
```diff
@@ -149,6 +149,10 @@ describe("Custom Type Builder", () => {
             },
           },
         },
+        // @ts-expect-error TS2739: Type '{ manifest: { apiEndpoint: string; }; }' is missing the following properties from type 'FrontEndEnvironment': repo, packageManager, supportsSliceSimulator, endpoints
```
Comment (xrutayisire) [inline packages/slice-machine/test/pages/custom-types.test.tsx:152]: ❓ Why now we have all of these TS errors?

File: packages/slice-machine/test/pages/slices.test.tsx (hunk 2 above)
Comment (xrutayisire) [inline packages/slice-machine/test/pages/slices.test.tsx]: ⚠ Why using data-cy instead of getByText? We should try removing dependencies in unit test to Cypress selectors as much as possible.

File: packages/slice-machine/test/pages/slices.test.tsx [hunk 14]
```diff
@@ -351,9 +352,14 @@ describe("slices", () => {
...
+      '[data-cy="create-slice-modal"]'
+    );
     await act(async () => {
-      fireEvent.click(submitButton);
+      if (createSliceModal) {
```
Comment (xrutayisire) [inline packages/slice-machine/test/pages/slices.test.tsx]: ⚠ Because of the `if`, this test seems not useful anymore, no? Meaning, if you don't find the modal, just do nothing.

We should assess before that the element exist at least I guess

**Review by xrutayisire — CHANGES_REQUESTED** [review]
First, quick review of the code. Super nice changes.
I will now try to find UI bugs also knowing the code now.

A lot of things done, knowing a bit more the changes, I think it was possible to split that, but we could come back to that later as a "How do create small PR?" meeting. :)

File: packages/slice-machine/test/pages/slices.test.tsx (hunk 2 above)
Comment (bapmrl) [reply packages/slice-machine/test/pages/slices.test.tsx]: I've fixed it by using [getByLabelText](https://testing-library.com/docs/queries/bylabeltext).

Comment (bapmrl) [reply packages/slice-machine/test/pages/slices.test.tsx]: I've fixed it by using [getByLabelText](https://testing-library.com/docs/queries/bylabeltext).

File: packages/slice-machine/test/pages/slices.test.tsx (hunk 14 above)
Comment (bapmrl) [reply packages/slice-machine/test/pages/slices.test.tsx]: I've fixed it by using [getByLabelText](https://testing-library.com/docs/queries/bylabeltext).

File: packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx (hunk 3 above)
Comment (bapmrl) [reply packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx:62]: I don't understand what you would like to know Marc. I didn't change anything here 🙂 I just had to wrap `Box` in `AppLayout` and `AppLayoutContent` as the layout isn't applied from the top level of the app anymore.

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx (hunk 4 above)
Comment (bapmrl) [reply packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:331]: Look for `@ts-expect-error` in the code and you'll see it's used extensively in our unit tests to provide incomplete data to Redux.

File: packages/slice-machine/test/pages/custom-types.test.tsx (hunk 13 above)
Comment (bapmrl) [reply packages/slice-machine/test/pages/custom-types.test.tsx:152]: Look for `@ts-expect-error` in the code and you'll see it's used extensively in our unit tests to provide incomplete data to Redux.

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 6 above)
Comment (bapmrl) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:30]: I think so. It's just code I've moved.

File: packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx (hunk 7 above)
Comment (bapmrl) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx:186]: If I simply set `viewBox` to `0 0 24 24` it won't work. Benjamin needs to update the icons in Figma to accommodate the smaller `viewBox`.
Screenshot 2023-08-03 at 14 46 28

File: packages/slice-machine/components/Tooltip/TextWithTooltip.tsx (hunk 9 above)
Comment (bapmrl) [reply packages/slice-machine/components/Tooltip/TextWithTooltip.tsx:54]: Maybe, but this line of code isn't part of the changes introduced by this pull request.

File: packages/slice-machine/components/Changelog/index.tsx (hunk 10 above)
Comment (bapmrl) [reply packages/slice-machine/components/Changelog/index.tsx]: We're using Prettier to avoid discussing formatting in pull requests.

However, I can add an additional `showReleaseWarning` variable if you think it's better?

```
useEffect(() => {
  setSelectedVersion(latestVersion);
}, [latestVersion]);

const showReleaseWarning =
  changelog.sliceMachine.versions.length === 0 || !selectedVersion;

return (
  <AppLayout>
    <AppLayoutContent>
      <BaseStyles sx={{ display: "flex", minWidth: 0 }}>
        {!isChangelogLoading ? (
          <>
            <Navigation
              changelog={changelog}
              selectedVersion={selectedVersion}
              selectVersion={(version) => setSelectedVersion(version)}
            />

            {showReleaseWarning ? (
              <Flex sx={{ paddingLeft: "32px" }}>
                <ReleaseWarning />
              </Flex>
            ) : (
              <VersionDetails
                changelog={changelog}
                selectedVersion={selectedVersion}
                packageManager={packageManager}
              />
            )}
          </>
        ) : (
          <ProgressCircle />
        )}
      </BaseStyles>
    </AppLayoutContent>
  </AppLayout>
);
```

File: packages/slice-machine/components/AppLayout/index.tsx (hunk 5 above)
Comment (bapmrl) [reply packages/slice-machine/components/AppLayout/index.tsx:16]: It could be either `otherProps` or `restProps`, but not the other options as we sometimes need to reference the unspread `props` like this:
```
const Component: FC<ComponentProps> = (props) => {
  const { someProp, ...otherProps } = props
  // ...
}
```

Personally, I prefer `otherProps`.

File: packages/slice-machine/components/Simulator/index.tsx (hunk 8 above)
Comment (bapmrl) [reply packages/slice-machine/components/Simulator/index.tsx:71]: Yes 🙂 https://prismic-team.slack.com/archives/C0441S8SKQR/p1690892860525849.

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 11 above)
Comment (bapmrl) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts:39]: Sadly, `paddingBottom` doesn't work when the component (in our case, `SideNav`) is used within a CSS grid (in our case, `PageLayoutPane`) 😕

File: packages/slice-machine/components/AppLayout/index.tsx (hunk 5 above)
Comment (xrutayisire) [reply packages/slice-machine/components/AppLayout/index.tsx:16]: Thanks, I will write a small thing for "otherProps" in code guidelines.
cc @MarcMcIntosh @hypervillain

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 12 above)
Comment (bapmrl) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts]: Yes. The purpose of `marginLeft` is to create space with the repository info 🙂
Screenshot 2023-08-03 at 15 17 57

File: packages/slice-machine/components/Simulator/index.tsx (hunk 8 above)
Comment (xrutayisire) [reply packages/slice-machine/components/Simulator/index.tsx:71]: Nice thanks!!

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 11 above)
Comment (xrutayisire) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts:39]: Ok I see :/

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 12 above)
Comment (xrutayisire) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts]: Yes but the removal of margin-right changed the design, no?

File: packages/slice-machine/test/pages/custom-types.test.tsx (hunk 13 above)
Comment (xrutayisire) [reply packages/slice-machine/test/pages/custom-types.test.tsx:152]: Ok yes for sure, but why now and not before here, strange no?

Comment (bapmrl) [reply packages/slice-machine/test/pages/custom-types.test.tsx:152]: Before, the `CreateCustomTypeBuilder` component being rendered in the tests wasn't including the `Navigation` menu. Now, we must provide some Redux data (i.e., `apiEndpoint`) so that the repository information within the menu can be rendered. However, I'm not providing all the data required by the Redux types so I resorted to adding a `@ts-expect-error` (like other places in the code).

Tell me if I'm not clear 🙂

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 12 above)
Comment (bapmrl) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts]: You're right, sorry! I'll fix that.

File: packages/slice-machine/test/pages/custom-types.test.tsx (hunk 13 above)
Comment (xrutayisire) [reply packages/slice-machine/test/pages/custom-types.test.tsx:152]: Ok I get it, thanks for this explanation!

File: packages/slice-machine/src/components/SideNav/SideNav.css.ts (hunk 12 above)
Comment (xrutayisire) [reply packages/slice-machine/src/components/SideNav/SideNav.css.ts]: Nice, I resolved!

**Review by xrutayisire — APPROVED** [review]
LGTM! Great work! It combines a lot of things that we want, and to continue the design change project it will now be simpler :)

File: packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/features/customTypes/customTypesTable/CustomTypesTablePage.tsx:62]: For the errors :)

File: packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx (hunk 7 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/SimulatorButton/index.tsx:186]: you can pass the `viewBox` as a prop and it should override it (although it might look weird since `veiwBox` basically sets the coordination upper and lower limits.

Are you sure it's not the width and height you want to change?

File: packages/slice-machine/components/Changelog/index.tsx (hunk 10 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/Changelog/index.tsx]: yes :) thanks

Comment (bapmrl) [reply packages/slice-machine/components/Changelog/index.tsx]: Done 🙂

Comment (MarcMcIntosh) [reply packages/slice-machine/components/Changelog/index.tsx]: thanks :)

**Review by MarcMcIntosh — APPROVED** [review]
